### ISSUE TITLE
configs not being saved

### ISSUE THREAD

**Issue by Kuzibaba** [lehjr/MachineMusePowersuits#84]
**Version info**
Java version: 8
Minecraft Forge version: 1.12.2 - 14.23.5.2860

**Describe the bug**
When I change a config in game (like activating fortune module) it doesn't get saved and I can't use it

**To Reproduce**
Just change a config and restart your game 

**Expected behavior**
For the config stay as I set it to be

**Mod list**
Numina 
modularpowersuits

**Log files**
(Link to pastebin, gist, or other hosting site preferred, file attachment okay.)
Game log (make sure it's the log that was generated during the gameplay the issue occurred in):
Crash report (if issue caused game to crash):
Server log 
**Screenshots**

**Additional context**

**Comment by lehjr** [comment]
This is probably something that will require quite a bit of effort to fix, specifically replacing Forge's sub-optimal config system with something like this: https://www.curseforge.com/minecraft/mc-mods/configuration
However, it makes little sense to put that much effort into a version nearing it's end of life (Forge for 1.12.2 has been available for 4 years now), especially since the documentation for the API in that version of the mod isn't even available anymore. Furthermore, introducing a hard dependency now for a version that's been out this long isn't exactly a popular idea either. Along those lines, the potential for introducing game breaking changes must also be evaluated, especially given Forge's questionable future now that all but one of the team members have moved on to their own project. This introduces the likelihood that development environments relying on Forge may cease to work, since they were never intended to work offline. So I have to weigh the potential for wasting effort that could be better spent on future versions rather than past ones. All that said, I have't ruled it out.

**Comment by Kuzibaba** [comment]
Yeah the forge's uncertain future is so sad but idk 1.12.2 seems like dead but i think it still has a solid community who is diehard 1.12.2(including me) so its kinda sad seeing some mods losing support for 1.12.2 while some didnt move on, so it creates a dilemma espacially for me who has a modpack full of 909 mods all handpicked. I nearly achieved my dream game but these support issues are just sad

**Comment by lehjr** [comment]
The thing to remember though is that the development environment is far more than just the jar that players download and without it working there are no more updates that can be done. As it is now, the one reason there are any updates possible is because Forge released an update to make their gradle backwards compatible. But the reality is that when mod authors focus too much on old version those mods get left behind and die out. Usually the only thing that keeps a mod alive is moving forward.

**Comment by lehjr** [comment]
As far as the config issue goes, newer versions don’t have that issue. After 1.12.2, the config system was completely overhauled. Actually, nearly everything after 1.12.2 is easier to work on for the most part.

**Comment by Kuzibaba** [comment]
Just asking, didnt you discover this bug at all,all the way back in 1.12.2 times

**Comment by lehjr** [comment]
Surprisingly no. However I did end up getting frustrated with things and took a break for awhile

**Comment by Kuzibaba** [comment]
Understandable, but its kinda weird NOBODY reported this issue lol

**Comment by Kuzibaba** [comment]
Isnt there a way for you to just set the modules which are not shoewng right now to be showing on default

**Comment by lehjr** [comment]
If I had to guess why the issue was never reported, on top of people clinging to outdated versions like 1.7.10, and a global pandemic, there’s also the issue of having to create a GitHub account. 

Fixing things in 1.12.2 isn’t trivial, especially if it involves the config system, a system so bad that Forge dropped it in the very next version of Minecraft. It would probably be easier to backport their new config system than to use the mess they have in 1.12.2

Do I invest time in a version where my development environment may suddenly stop working again letting all that work go to waste while I fall further behind, or do I keep working on 1.19.2 so I can get closer to an actual release before I move on to supporting Farbric/Quilt/NeoForged? I still don’t even have a system in place for mod integration yet and that won’t be a small task.

**Comment by Kuzibaba** [comment]
Keep moving on, but as i said, isnt there a way to set those on by default if there isnt than its okay
